### PR TITLE
Analytics for Builds, and other fixes

### PR DESCRIPTION
**Pull request by hansl**
Multiple changes:

1. Builder stats for every build. This was tested using logging and Architect CLI.  
    The only missing piece is for the Angular CLI to pass an instance of `Analytics` interface.
1. Validation script for the user analytics, to make sure the document is kept up to date.
1. Rework some metrics into dimensions, which lets us make better reports.
1. adding duration tracking for all commands.

### CODE REVIEW

File: packages/angular_devkit/build_angular/plugins/webpack/analytics.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+/**
+ * Faster than using a RegExp, so we use this to count occurences in source code.
+ * @param source The source to look into.
+ * @param match The match string to look for.
+ * @param wordBreak Whether to check for word break before and after a match was found.
+ * @return The number of matches found.
+ * @private
+ */
+export function countOccurrences(source: string, match: string, wordBreak = false): number {
```
Comment (alexeagle) [inline packages/angular_devkit/build_angular/plugins/webpack/analytics.ts]: would be nice to unit test this - in case we find some weird logging edge case

File: packages/angular_devkit/build_angular/src/utils/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export const build = {
```
Comment (alexeagle) [inline packages/angular_devkit/build_angular/src/utils/analytics.ts]: check if its used

File: packages/angular_devkit/build_angular/plugins/webpack/analytics.ts (hunk 1 above)
Comment (hansl) [reply packages/angular_devkit/build_angular/plugins/webpack/analytics.ts]: Done (and fixed a bug)

File: packages/angular_devkit/build_angular/src/utils/analytics.ts (hunk 2 above)
Comment (hansl) [reply packages/angular_devkit/build_angular/src/utils/analytics.ts]: Whole file removed.
